export-mock-ddi-data.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+usage() {
+cat <<EOF
+Export a CSV file of Drug-Drug interactions from the Pathways system.
+
+Writes out to <outfile>
+
+Resultant CSV file is of the format:
+  drug_a, drug_b, {agonism or harmful}, time, unit
+
+Usage:
+    $0 <agonism|harmful> <outfile>
+EOF
+}
+
+
+loading() {
+    echo 'Exporting'
+    while true; do
+        printf "\r | "
+        sleep 0.125
+        printf "\r / "
+        sleep 0.125
+        printf "\r - "
+        sleep 0.125
+        printf "\r \\ "
+        sleep 0.125
+    done
+}
+
+
+main() {
+    if [ $# -ne 2 ]; then
+        usage
+        exit 2
+    fi
+
+    local readonly flavour="$1"; shift
+    local readonly outfile="$1"; shift
+    case "$flavour" in
+        harmful|agonism)
+            loading &
+            local readonly LOAD_PID="$!";
+            trap  "kill $LOAD_PID 2>&1 > /dev/null; exit" SIGTERM SIGKILL SIGINT
+            disown
```

Comment:
The export takes 5-10 secs, which is a bit unacceptable to have no output at all; loading spinner is on its own thread

`$!` is the PID of the last forked process, `disown` removes the proc from the shell's job list (so it can't be `fg`d) after the script is dead.

`trap` sets a callback (passed as a string) to the given list of signals; so it kills the spinner proc when it's ctr-c'd/sent a kill signal/the parent proc dies